tests/modules/programs/tealdeer/default-settings.nix

```
@@ -0,0 +1,19 @@
+{ config, lib, ... }:
+let hmActivationDagResult = import ./dag-result.nix { inherit lib; };
+in {
+  config = {
+    programs.tealdeer = {
+      package = config.lib.test.mkStubPackage { name = "tldr"; };
+      enable = true;
+    };
+
+    home.file."tealdeer-test.txt".text =
+      hmActivationDagResult config.home.activation;
+
+    nmt.script = ''
+      assertFileExists home-files/tealdeer-test.txt
+
+      assertFileRegex "home-files/tealdeer-test.txt" '/nix/store/.*tealdeer.*/bin/tldr --update'
```

Not sure if there's a better way to test that activation script is there.

---

I think it should be sufficient to do

```suggestion
      assertFileRegex "activate" '/nix/store/.*tealdeer.*/bin/tldr --update'
```

---

Thanks! That's much simpler. 
But what magic is this? Where this `activate` is coming from? 😲

---

Is this the [main activation script](https://github.com/pedorich-n/home-manager/blob/8a39882f926e68840bf1131f4b6740d1b0ab4a89/modules/home-environment.nix#L713), but in tests `$out` is `/`?

---

Yes, paths in the tests are relative the built activation package. So, basically the output of `home-manager build`, You'll notice that there is an `activate` file in there, as well as a link `home-files`, that typically is followed in tests.